text/2020-09-03-factored-prefix.md

```diff
@@ -0,0 +1,71 @@
+# Factored Key Prefix
+
+## Summary
+
+TiKV provides a facility where the client can set a key prefix that will be used for all keys in its requests.
+
+## Motivation
```

Comment:
If it is use to support multi-tenancy, how does it compare to separating users on different TiKV processes? With multiple processes, we have tools for resource isolation, like systemd to limit CPU, memory and IO, and TiKV configs to limit thread-pool and IO usage on application level. When we mixed data from different users in one single TiKV instance, those tools are not available or hard to implement. But of cause it might make it hard for users to share resources.

Reply:
This is better discussed in the key spaces RFC. But separate TiKV processes doesn't scale down well. The approach provided by these RFCs scales down to one region (100 MB).

Reply:
yeah, seems that the crucial question is what's the smallest dataset per tenant we want to support.